Source/Events/EventConverter.cs

```diff
@@ -27,69 +26,231 @@ public EventConverter(IEventTypes eventTypes)
         {
-            var eventType = source.Type.To<EventType>();
-            var clrEventType = _eventTypes.GetTypeFor(eventType);
-            try
+            events = default;
+
+            if (source == null)
             {
-                var content = JsonConvert.DeserializeObject(source.Content, clrEventType);
-                if (source.External)
+                error = new ArgumentNullException(nameof(source));
+                return false;
+            }
+
+            var list = new List<CommittedEvent>();
+            foreach (var sourceEvent in source)
+            {
+                if (!TryToSDK(sourceEvent, out var @event, out error))
                 {
-                    return new CommittedExternalEvent(
-                        source.EventLogSequenceNumber,
-                        source.Occurred.ToDateTimeOffset(),
-                        source.EventSourceId.To<EventSourceId>(),
-                        source.ExecutionContext.ToExecutionContext(),
-                        eventType,
-                        content,
-                        source.Public,
-                        source.ExternalEventLogSequenceNumber,
-                        source.ExternalEventReceived.ToDateTimeOffset());
+                    return false;
                 }
                 else
                 {
-                    return new CommittedEvent(
-                        source.EventLogSequenceNumber,
-                        source.Occurred.ToDateTimeOffset(),
-                        source.EventSourceId.To<EventSourceId>(),
-                        source.ExecutionContext.ToExecutionContext(),
-                        eventType,
-                        content,
-                        source.Public);
+                    list.Add(@event);
                 }
             }
-            catch (Exception ex)
-            {
-                throw new CouldNotDeserializeEvent(
-                    source.Type.Id.To<EventTypeId>(),
-                    clrEventType,
-                    source.Content,
-                    source.EventLogSequenceNumber,
-                    ex);
-            }
+
+            events = new CommittedEvents(list);
+            error = null;
+            return true;
         }
 
         /// <inheritdoc/>
         public CommittedEvents ToSDK(IEnumerable<Contracts.CommittedEvent> source)
-            => new CommittedEvents(source.Select(ToSDK).ToList());
+            => TryToSDK(source, out var events, out var error) ? events : throw error;
+
+        /// <inheritdoc/>
+        public bool TryToSDK(Contracts.CommitEventsResponse source, out CommitEventsResult result, out Exception error)
+        {
+            result = default;
+
+            if (!TryToSDK(source.Events, out var events, out error))
+            {
+                return false;
+            }
+
+            error = null;
```

Comment:
I meant to say if TryToSDK is true

Reply:
Just got an idea here, might not make sense. 

But what if you had an inner class called ToSDK and one that was called ToProtobuf, and on it you had Try() methods. 

Then it would read like this:

```csharp
ToSDK.Try(....)
```

Reply:
I don't really see the benefits with that approach. Care to elaborate?

Reply:
Done